### ISSUE TITLE
Chore still not included in changelog

### ISSUE THREAD

**Issue by jeremybdk** [mathieudutour/github-tag-action#54]
Hello Mathieu,

Sorry for opening a new issue, but I cannot reopen the older one,
I still cannot get chores to be added to the changelog even using v5.1
Is there anything I should do/setup to have them displayed in the changelog?

Thanks a lot,

_Originally posted by @mathieudutour in https://github.com/mathieudutour/github-tag-action/issues/50#issuecomment-753875661_

**Comment by koiralakiran1** [comment]
@mathieudutour This is a followup comment for https://github.com/mathieudutour/github-tag-action/issues/52#issuecomment-755495029

Instead of creating a new issue, I thought I'll just comment here.
I think there's something breaking in the latest version. I've tried everything to get it to bump the non-default (for me, patch) version but it doesn't work. I don't have any existing releases / tags.

Here's my workflow:
![image](https://user-images.githubusercontent.com/11130898/103915140-eb21af00-5132-11eb-8347-97b1b8f8890a.png)

I pushed to branch `live-test` with commit `git commit -m "feat: test commit"`

I thought it would bump the `minor` version but it still bumps the `patch` version.

**Comment by mathieudutour** [comment]
it'd be useful if you could link to your repo where the workflow is failing. It's working fine on mine: https://github.com/mathieudutour/test-bump/releases

**Comment by jeremybdk** [comment]
Hey @mathieudutour what about commit with "chore(test_chore): Add to Changelog" ? that's mainly my issue, but my repo is private.

**Comment by jeremybdk** [comment]
I ended up forking your repo and recreating my issue: https://github.com/jeremybdk/test-bump
You can see that chore will not be included in the changelog,

Thanks

**Comment by koiralakiran1** [comment]
Seems like it doesn't work correctly on a branch except for the repo's default branch (not sure if it's the expected behavior):
https://github.com/koiralakiran1/test-bump/runs/1666433143

I created a test branch `main-test` from `main` (i.e default branch) and included it in `release_branches` and pushed a similar  (`feat`) commit as I pushed in `main`. It bumped the minor in `main` but still bumps the `patch` in `main-test`.

**Comment by mathieudutour** [comment]
ah interesting. It might be because of this: https://github.com/mathieudutour/github-tag-action/blob/master/src/github.ts#L33. I think we should get the sha of head of the branch you're on

**Comment by koiralakiran1** [comment]
> ah interesting. It might be because of this: https://github.com/mathieudutour/github-tag-action/blob/master/src/github.ts#L33. I think we should get the sha of head of the branch you're on

Glad to know that what I suspected was right. I was just looking into the code the other day :)

**Comment by jeremybdk** [comment]
Hello @mathieudutour sorry to keep bothering you.
Any plan to fix the chore issue that prevents chore commits to be included in the changelog?
It was also referenced here https://github.com/mathieudutour/github-tag-action/issues/17 and was tagged as fixed but it seems that it's still broken.

Thank you so much

**Comment by mathieudutour** [comment]
I don't have time to work on this sorry, but I'm happy to review a PR

**Comment by jeremybdk** [comment]
> I don't have time to work on this sorry, but I'm happy to review a PR

Thanks will try my best :)

**Comment by jeremybdk** [comment]
Hello @koiralakiran1,
Is this suppose to fix my original issue here or only the one you raised?

Thanks

**Comment by koiralakiran1** [comment]
Hi @jeremybdk,

I'm extremely sorry that I closed this issue without taking into consideration the original issue that you raised. The changes I made are only going to fix the issue that I raised in the comment above. I just tested the latest version and your issue still persists.

Investigating your issue, I see that the `custom_release_rules` that we pass to the action is only used for bumping the version but never used to generate the changelog. I think somehow we should pass this information as an appropriate config to `generateNotes` on line 149 below:
https://github.com/mathieudutour/github-tag-action/blob/5961cf9c4dfdab0d0ca4bab9f1d2403960331ef8/src/action.ts#L148-L149

(which comes from [release-notes-generator](https://github.com/semantic-release/release-notes-generator/blob/master/index.js#L29))

An alternative approach would be to allow configuring both `release-notes-generator` & `commit-analyzer` using [semantic rule configuration file](https://github.com/semantic-release/release-notes-generator#usage) and passing these appropriate configs to `generateNotes` but I'm currently not sure how to go about doing these changes to make it work.

Thanks
